### ISSUE TITLE
Any plans to migrate to vite or others?

### ISSUE THREAD

**Issue by moonlt5** [electron-react-boilerplate/electron-react-boilerplate#3188]

**Comment by amilajack** [comment]
I'm considering integrating electron-next. Biggest blocker here is mostly time.

**Comment by kbrisso** [comment]
> I'm considering integrating electron-next. Biggest blocker here is mostly time.

I thought Next was used for mobile apps and low bandwidth?

**Comment by amilajack** [comment]
Not exactly. At its core, next.js is a set of optimal webpack configs optimized for react ssr/isr/csr.

If we migrate to electron-next, we'll be trading some flexibility and simplicity for better performance. ERB has no solution for ssr at the moment.

**Comment by BangHyeongSik** [comment]
If so, is it 5 version of ERB?

**Comment by oliverpatrick** [comment]
> Not exactly. At its core, next.js is a set of optimal webpack configs optimized for react ssr/isr/csr.
> 
> If we migrate to electron-next, we'll be trading some flexibility and simplicity for better performance. ERB has no solution for ssr at the moment.

Wouldn't it be better to fork the project and start an electron-next-boilerplate instead of disrupting the vanilla react boilerplate?

**Comment by ivanjeremic** [comment]
I would stick to plain react no next and use Vite.js instead of webpack.

**Comment by amilajack** [comment]
> I would stick to plain react no next and use Vite.js instead of webpack.

I like this idea, specifically migrating to vite or esbuild. Migrating to next will only increase the already high barrier to entry.

**Comment by aguynamedben** [comment]
+1 Vite looks pretty sick, has a ton of docs, and has advanced technical breakdowns on how it does things.

I'm also looking at Tauri for fun... not sure that's a realistic switch up for my team in the short term, but Tauri + Vite seems like a positive development in our world!

**Comment by aguynamedben** [comment]
> specifically migrating to vite or esbuild

FWIW Vite uses esbuild under the hood

**Comment by ivanjeremic** [comment]
Yes like someone said before Vite uses ESBuild under the hood, and there are so many projects moving to it now and not because it's the hipp new thing it's because it is do damn fast and gives you such a good dev experience, the cool thing about this is this can turn into a boilerplate for every frontend framework and not only react since vite can handle them all.

**Comment by andrewschreiber** [comment]
I realized it would be easier to migrate my project from webpack to rspack instead of webpack to vite. Rspack (https://rspack.dev/) is meant to be a 10x faster ~drop-in replacement for webpack, so I'm putting together a package that will be called `electron-rspack` that will make this process as fast and easy as possible.

**Comment by ivanjeremic** [comment]
> I realized it would be easier to migrate my project from webpack to rspack instead of webpack to vite. Rspack (https://rspack.dev/) is meant to be a 10x faster ~drop-in replacement for webpack, so I'm putting together a package that will be called `electron-rspack` that will make this process as fast and easy as possible.
> 
> Will create a PR here when it's ready!

I think you missed the development of rolldown which is faster then rspack and generally cool things happening soon with Vite.js, here more [EvanYou Talk](https://www.youtube.com/watch?v=EKvvptbTx6k&t=78s) Also it is about DX and vite is unmatched here!
